src/finch/interface/lazy.py

```diff
@@ -1660,3 +1685,63 @@ def std(
     x = defer(x)
     d = var(x, axis=axis, correction=correction, keepdims=keepdims)
     return pow(d, 0.5)
+
+
+def argmin(x, /, *, axis: int | None = None, keepdims: bool = False, init=None):
+    """
+    Returns the indices of the minimum values in input array ``x`` along given axis.
+    """
+    if isinstance(axis, tuple):
+        raise ValueError("Type of axis should is only allowed to be int or None.")
+
+    x = defer(x)
+    computed_x = _compute(x)
+    if hasattr(computed_x, "to_numpy") and callable(computed_x.to_numpy):
```

Comment:
You shouldn't need to compute the arrays. This should be a fusable operation. the whole operation should be one line, as so:

`elementwise(first, reduce(minby, elementwise(Pair, A, IndicesTensor(A.size))))`.

Construct a lazy IndicesTensor using Osman's approach, like ConcatTensor.